tap_polarsh/tap.py

```diff
@@ -76,6 +83,13 @@ def discover_streams(self) -> list[Stream]:
                 "components": openapi_schema["components"],
             }
             resolved_schema = resolve_schema_references(schema)
-            streams.append(stream_type(tap=self, schema=resolved_schema))
+            streams_dict[stream_type.name] = stream_type(
+                tap=self,
+                schema=resolved_schema,
+            )
 
-        return sorted(streams, key=lambda x: x.name)
+        for stream_name, stream_type in streams_dict.items():
+            if stream_name.startswith("benefits_"):
+                stream_type.child_streams.append(streams_dict["benefit_grants"])
```

Comment:
This actually doesn't work because of

```python
            if stream.parent_stream_type:
                self.logger.debug(
                    "Child stream '%s' is expected to be called "
                    "by parent stream '%s'. "
                    "Skipping direct invocation.",
                    type(stream).__name__,
                    stream.parent_stream_type.__name__,
                )
                continue
```

https://github.com/meltano/sdk/blob/1f401b90acb7f75296c5b800936625ffd23c9a47/singer_sdk/tap_base.py#L480-L488

Reply:
https://github.com/meltano/sdk/discussions/2846